contrib/gen_manual/Makefile

```diff
@@ -56,8 +56,9 @@ endif
 .PHONY: default
 default: gen_manual
 
+CXX_FOR_BUILD = $(CXX)
```

Comment:
I suspect what you meant here is : 
```
CXX_FOR_BUILD ?= $(CXX)
```

This way, `CXX_FOR_BUILD` won't be redefined if it's already set via an environment variable.